docs/src/docs/asciidoc/user/resilience.adoc

```diff
@@ -0,0 +1,189 @@
+latest value.
+
+In particular, it means that as soon as an update from the server might be missed, we need to clear
+the caching tier. It also means that if we fail to set a value to the authoritative tier, we can't use
+the cache anymore. On this key at least.
+
+Eventual means the correct value will eventually be there. So if we loose contact with the server, it is possible to keep
+using a possibly stale cache entry. This means that an client with eventual consistency shouldn't need to clear it's cache
+at all. However, it means two things:
+
+* We need to be able to update the caching tier in that case
+* We need to get updates from the server when the situation resumes
+
+=== Being too eventual
+
+We can argue that if a client stays disconnected for too long, it means the caching tier content is now too obsolete. There
+are two ways to handle that
+
+* Expiration should take care of it. So entries should have an expiration set to their expected time validity.
+* We configure "too long" which will clear the cache
+
+== Clustering
+
+When using a distributed cache, your Ehcache client will connect to a remote Terracotta server (or multiple in case
+of stripping) which act as the authoritative tier.
+
+This server data can be replicated to other servers called mirrors. How your client will handle being cut from
+the cluster is defined in `ClusteringServiceConfiguration`.
+
+=== Timeouts
+
+If a client doesn't receive an answer in the configured time, it will timeout. This can happen for different reasons
+including a network cut, a long full GC or a server down.
+
+Three different types of timeouts have been defined.
+
+* Read: Operations reading data from the server
+* Mutative: Operations modifying data on the server
+* Lifecycle: Operations related to the lifecycle like creating and destroying a cache or simply connecting to the server
+
+By default, all are set to 5 seconds.
+
+=== Retries
+
+When a client timeouts, it might retry right away. That are two parameters for that:
+
+* Number of retries (default is 0)
+* Sleep time between retries (default is 0)
+
+Note that for each retry, the timeout will be enforced. So, far example, 3 retries with a timeout of 5 seconds and a sleep
+of 1 second will mean 20 seconds before returning.
+
+*Note to myself:* Should the retry be on the resilience strategy?
+
+=== In case of failure
+
+If we failed to get an answer, the cache [#Resilience Strategy] will kick in and return whatever it was configured to return.
+
+Resilience strategy is not used for lifecycle events. So in this case you will get a `StoreAccessException`.
+
+=== After a failure
+
+We will now consider a server with no mirror.
+
+The first thing that can happen is a server that has problem answering correctly all the time but that is still there.
+In this case, after 10 timeouts (configurable), the client will consider the server down to prevent waiting on requests.
```

Comment:
Isn't this a matter of tuning timeouts correctly, if the cache sometimes answers correctly why would I want to give up on it completely.  If the connection is completely down then I understand not waiting for timeouts and going straight to resilience strategy, but to me making that decision only makes sense when you *know* that the timeout will happen, not when you think there is some significant chance it could happen.

Reply:
Yes. If it's erratic, it's better to try. If it's too erratic or down, it's better to try in background and answer locally to prevent the client from waiting and go to degraded mode.